### PR TITLE
Cleaned up auth documentation

### LINKED ISSUE #759: $getAuth(), provider is undefined
### Version info

**Angular:** 1.5.6

**Firebase:**3.0.3

**AngularFire:** 2.0.1

**Other (e.g. Node, browser, operating system) (if applicable):** Ionic

### Test case

The doc says that $getAuth() returns :

> uid, provider (string identifying the provider), auth (the authentication token payload), and expires (expiration time in seconds since the Unix epoch)

But

`var authData = $scope.authObj.$getAuth();`
authData.uid is okay, but authData.provider and authData.auth are undefined.

### Steps to reproduce

```
var authData = $scope.authObj.$getAuth();
var provider = authData.provider
var uid = authData.uid
```
### Expected behavior

If the doc is correct, provider should be a string, and it is currently "undefined"

### Actual behavior

provider is "undefined".

### PR DESCRIPTION
**Pull request by jwngr**
Resolves #759 and #760. I also realized a ton of other things in the docs had the wrong signature or were named incorrectly, so I tried cleaning up as much of that as I could. I pretty much read through every auth-related thing and tried to get it updated.

### CODE REVIEW

File: docs/guide/user-auth.md [hunk 1]
```diff
@@ -225,7 +223,28 @@ it will contain the following attributes:
 | Key | Description |
 |-----|-------------|
 | `uid` |	A user ID, guaranteed to be unique across all providers. |
-| `provider` | The authentication method used (e.g. "anonymous" or "google"). |
+| `provider` | The authentication method used (e.g. "anonymous" or "google.com"). |
+| `token` | The contents of the authentication token (an OpenID JWT). |
+
+The contents of `auth.token` will further contain the following information:
```
Comment (katowulf) [inline docs/guide/user-auth.md]: <del>further</del>

File: docs/reference.md [hunk 2]
```diff
@@ -754,41 +752,35 @@ offAuth();
 
 ### $signOut()
 
-Unauthenticates a client from the Firebase database. It takes no arguments and returns no value. When logout is called, the
-`$onAuthStateChanged()` callback(s) will be fired.
+Unauthenticates a client from the Firebase Database. It takes no arguments and returns no value.
+When called, the `$onAuthStateChanged()` callback(s) will be fired.
```
Comment (katowulf) [inline docs/reference.md]: Not your change, but suggestion:  "... will be triggered/invoked."

File: docs/reference.md [hunk 3]
```diff
@@ -711,12 +709,12 @@ time in seconds since the Unix epoch) - and more, depending upon the provider us
 will be returned. Otherwise, the return value will be `null`.
 
 ```js
-var authData = $scope.authObj.$getAuth();
+var firebaseUser = $scope.authObj.$getAuth();
```
Comment (katowulf) [inline docs/reference.md]: Hrm, did we change this to $getCurrentUser() in another method? It's pretty strange construct right now, with this revision:  "the firebase user is equal to the auth object dot get auth"

**Comment by katowulf** [comment]
A couple nits to consider. Bin if they are annoying.

File: docs/reference.md (hunk 3 above)
Comment (jwngr) [inline docs/reference.md]: So the `getCurrentUser()` is what it is called in the new Firebase `3.x.x` SDK. We chose to keep the `$getAuth()` name in AngularFire although we may have been better off using `$getCurrentUser()`. I added [a note](https://github.com/firebase/angularfire/issues/764#issuecomment-224369132) to #764 to consider this change.

**Comment by jwngr** [comment]
Back to you @katowulf.

**Comment by katowulf** [comment]
☃☃☃  Are we waiting on Abe to lgtm? If not, merge away.

**Comment by jwngr** [comment]
Nope, just needed either of you. Merged.

**Pull request merged**